Source/Core/DolphinQt2/Debugger/MemoryWidget.cpp

```diff
@@ -0,0 +1,575 @@
+{
+  Settings::Instance().SetMemoryVisible(false);
+}
+
+void MemoryWidget::Update()
+{
+  m_memory_view->Update();
+}
+
+void MemoryWidget::LoadSettings()
+{
+  QSettings settings;
+
+  bool search_ascii = settings.value(QStringLiteral("memorywidget/searchascii"), true).toBool();
+  bool search_hex = settings.value(QStringLiteral("memorywidget/searchhex"), false).toBool();
+
+  m_find_ascii->setChecked(search_ascii);
+  m_find_hex->setChecked(search_hex);
+
+  bool type_u8 = settings.value(QStringLiteral("memorywidget/typeu8"), true).toBool();
+  bool type_u16 = settings.value(QStringLiteral("memorywidget/typeu16"), false).toBool();
+  bool type_u32 = settings.value(QStringLiteral("memorywidget/typeu32"), false).toBool();
+  bool type_float = settings.value(QStringLiteral("memorywidget/typefloat"), false).toBool();
+  bool type_ascii = settings.value(QStringLiteral("memorywidget/typeascii"), false).toBool();
+
+  m_type_u8->setChecked(type_u8);
+  m_type_u16->setChecked(type_u16);
+  m_type_u32->setChecked(type_u32);
+  m_type_float->setChecked(type_float);
+  m_type_ascii->setChecked(type_ascii);
+
+  bool bp_rw = settings.value(QStringLiteral("memorywidget/bpreadwrite"), true).toBool();
+  bool bp_r = settings.value(QStringLiteral("memorywidget/bpread"), false).toBool();
+  bool bp_w = settings.value(QStringLiteral("memorywidget/bpwrite"), false).toBool();
+  bool bp_log = settings.value(QStringLiteral("memorywidget/bplog"), true).toBool();
+
+  m_bp_read_write->setChecked(bp_rw);
+  m_bp_read_only->setChecked(bp_r);
+  m_bp_write_only->setChecked(bp_w);
+  m_bp_log_check->setChecked(bp_log);
+}
+
+void MemoryWidget::SaveSettings()
+{
+  QSettings settings;
+
+  settings.setValue(QStringLiteral("memorywidget/searchascii"), m_find_ascii->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/searchhex"), m_find_hex->isChecked());
+
+  settings.setValue(QStringLiteral("memorywidget/typeu8"), m_type_u8->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/typeu16"), m_type_u16->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/typeu32"), m_type_u32->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/typeascii"), m_type_ascii->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/typefloat"), m_type_float->isChecked());
+
+  settings.setValue(QStringLiteral("memorywidget/bpreadwrite"), m_bp_read_write->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/bpread"), m_bp_read_only->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/bpwrite"), m_bp_write_only->isChecked());
+  settings.setValue(QStringLiteral("memorywidget/bplog"), m_bp_log_check->isChecked());
+}
+
+void MemoryWidget::OnTypeChanged()
+{
+  bool type_u8 = m_type_u8->isChecked();
+  bool type_u16 = m_type_u16->isChecked();
+  bool type_u32 = m_type_u32->isChecked();
+  bool type_ascii = m_type_ascii->isChecked();
+
+  MemoryViewWidget::Type type;
+
+  if (type_u8)
```

Comment:
Is there any reason for defining all of the `bool`s first, and not doing `if (m_type_u8->isChecked())`?